src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -52,35 +49,23 @@ public Tags(string[] keys, object[] values)
 
         public readonly bool Equals(Tags other)
         {
-            // Equality check for Keys
-            // Check if the two string[] are equal
-            var keysLength = this.Keys.Length;
+            var length = this.KeyValuePairs.Length;
 
-            if (keysLength != other.Keys.Length)
+            if (length != other.KeyValuePairs.Length)
             {
                 return false;
             }
 
-            for (int i = 0; i < keysLength; i++)
+            for (int i = 0; i < length; i++)
             {
-                if (!this.Keys[i].Equals(other.Keys[i], StringComparison.Ordinal))
+                // Equality check for Keys
+                if (!this.KeyValuePairs[i].Key.Equals(other.KeyValuePairs[i].Key, StringComparison.Ordinal))
```

Comment:
Here we look-up into the array twice for each side. Might be slightly faster to take pointers like...

```csharp
ref var left = ref this.KeyValuePairs[i];
ref var right = ref other.KeyValuePairs[i];
```

...and then use those to compare the key & value.